### ISSUE TITLE
Alternativas de login para comentar/enviar propostas

### ISSUE THREAD

**Issue by rsip22** [tmadeira/dolores#1]
O site deveria ter opções alternativas de login para quem quer comentar e/ou enviar propostas e não tem conta no Google ou Feissebuqui, nem que seja um OpenID da vida. O fato de não ter limita a participação e, aliás, não sei como essa exigência de conta daquelas duas empresas para se envolver em projeto do partido poderia ser compatível com as filosofias do PSOL.

**Comment by tmadeira** [comment]
Não tenho como mexer nisso agora, mas se alguém quiser enviar um patch será muito bem-vindo.

**Comment by rsip22** [comment]
Já que é feito em wordpress, só instalar o plugin do OpenID não adianta?

https://wordpress.org/plugins/openid/

**Comment by tmadeira** [comment]
Não, porque os fluxos de cadastro/autenticação que fizemos especialmente para o Dolores não funcionariam se a autenticação acontecesse numa página própria. Acontecem via JavaScript quando você clica em qualquer coisa que precise de cadastro (por exemplo, tenta curtir alguma coisa ou escrever um comentário). A adaptação teria que levar esses fluxos em conta.

**Comment by hillarysousa** [comment]
Será que não valeria a pena ativar a opção de cadastro no painel administrativo do Wordpress para então tratar um formulário de login integrado? Assim ele poderia funcionar com os popups de login que aparecem ao tentar interagir com o site

**Comment by tmadeira** [comment]
Creio que implementar autenticação por login/senha nos popups de login do site já seja suficiente... Nem precisaria fazer um fluxo de cadastro diferente em outro lugar.

Acho que as duas coisas que precisariam ser pensadas pra além dessa autenticação, num segundo momento, seriam imagem de perfil (que hoje é baixada diretamente de Google ou Facebook) e confirmação de e-mail (hoje só pegamos dessas outras redes).

**Comment by hillarysousa** [comment]
bom, o wordpress tem integração com o gravatar. ele pode puxar a imagem de perfil de lá. já a confirmação de e-mail o próprio wordpress envia quando se cria a conta através do painel

**Comment by tmadeira** [comment]
Perfeito. Isso torna as coisas do segundo momento menos prioritárias. De qualquer forma, o importante agora é colocar autenticação por login/senha na popup. Precisa de mudanças no componente React e no endpoint da API.

**Comment by hillarysousa** [comment]
bom, eu ainda tô estudando React, mas posso ajudar em alguma coisa se precisar xD

**Comment by tmadeira** [comment]
@hillarysousa Esse é o principal componente que precisa ser editado: https://github.com/tmadeira/dolores/blob/master/src/js/shared/components/Authenticator.react.js A lógica do React não é muito difícil.

Do lado do servidor, tem que modificar as funções de https://github.com/tmadeira/dolores/blob/master/src/php/dlib/users.php (que são chamadas por https://github.com/tmadeira/dolores/blob/master/src/php/dlib/api/DoloresSigninAPI.class.php), em particular `authenticate`.

Eu tô sem tempo pra fazer isso, infelizmente, mas se você (ou alguém que estiver lendo) quiser pegar essa tarefa posso orientar qualquer detalhe!

**Comment by hillarysousa** [comment]
vou dar uma olhada nesses arquivos e ver se consigo fazer algo. qualquer coisa grito por aqui
